tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1420,10 +1602,21 @@ void CommandQueue::wait_until_empty() {
 
 void CommandQueue::set_mode(const CommandQueueMode& mode_) {
     TT_ASSERT(not this->trace_mode(), "Cannot change mode of a trace command queue, copy to a non-trace command queue instead!");
+    if (this->mode == mode_) {
+        // Do nothing if requested mode matches current CQ mode.
+        return;
+    }
     this->mode = mode_;
     if (this->async_mode()) {
-        this->start_worker();
-    } else {
+        num_async_cqs++;
+        num_pt_cqs--;
```

Comment:
If you look at this out of context, you don't know what pt means. Maybe just expand to full name